src/uct/ib/base/ib_device.c

```diff
@@ -646,6 +646,41 @@ uint8_t uct_ib_to_fabric_time(double time)
     }
 }
 
+uint8_t uct_ib_to_rnr_fabric_time(double time)
+{
+    const uint8_t max_rnr = 32;
+    double rnr_time_ms[max_rnr];
+    double avg_ms;
+    uint8_t t;
+
+    for (t = 0; t < max_rnr; t++) {
+        if (t >= 4) {
+            if (ucs_is_even(t + 1)) {
```

Comment:
may be simpler ```!ucs_is_even(t)```

Reply:
good catch! done

added `ucs_is_odd` and used it here